### PR TITLE
update slot to fix detach

### PR DESCRIPTION
**Pull request by hchoura**

### CODE REVIEW

File: src/foam/core/Slot.js [hunk 1]
```diff
@@ -642,7 +642,7 @@ foam.CLASS({
       if ( arguments.length != 1 ) return this.SUPER.apply(this, arguments);
       var subs = this.slots.map(s => s.sub(l));
```
Comment (kgrgreer) [inline src/foam/core/Slot.js:643]: Could we instead remove the edit on line 645 and instead change this line to:
var subs = this.slots.map(s => s && s.sub(l));

Then we don't waste space on null values?

File: src/foam/core/Slot.js [hunk 2]
```diff
@@ -557,7 +557,7 @@ foam.CLASS({
       this.cleanup_ = {
         detach: function() {
           for ( var i = 0 ; i < subs.length ; i++ ) {
-            subs[i].detach();
+            if ( subs[i] ) subs[i].detach();
```
Comment (kgrgreer) [inline src/foam/core/Slot.js:560]: How is this possible? Only if someone calls sub(null). Maybe better to override the sub() method and ignore null values there then remove this check. We could also log when it happens so developers would know to fix their problem.
